### ISSUE TITLE
Make Csound for iOS libraries dynamic

### ISSUE THREAD

**Issue by kunstmusik** [csound/csound#482]
- Rework build to generate libcsound and libndfile as shared libraries
- Package into Framework
- Rework examples project to use dynamic linking
- Update documentation for new linking strategy

This should clear up licensing issues yet still protect the LGPL nature of Csound.

**Comment by aure** [comment]
Stephane and I can help with this - it would be good to have this coordinated with AudioKit to be on the same code base again.  Stephane has improved the performance of the csound obj considerably.

**Comment by megastep** [comment]
Might want to take a look at the changes I made to achieve this on our branch at https://github.com/audiokit/csound

For CsoundObj, it could be a good idea to back-port the version we're using in AudioKit now. It's shared among both iOS and OS X.

**Comment by kunstmusik** [comment]
@aure @megastep  If you could submit pull-requests for CsoundObj that would certainly help.  For build changes to use dynamic libraries, I found some build scripts in a folder with audiokit in the name: was that what you were talking about?  (It looked like the primary build scripts within the iOS directory tree were still building static libraries)

**Comment by megastep** [comment]
Yes, I created a bunch of scripts (largely inspired by the existing installer scripts) to build binaries for the purpose of including in the frameworks that ship with AudioKit. These scripts build universal libraries for both OS X and iOS and update the binaries in the frameworks.

The main difference, at least for OS X, is that we use float samples instead of the default doubles, for consistency with the iOS side of things.

I have done something similar to `libsndfile` in our own fork as we also needed dynamic libraries to work around the LGPL. Conversely, our builds of Csound mostly only require libsndfile as a dependency.

I wouldn't mind contributing CsoundObj back to csound, however there'll probably be some reshuffling involved considering we merged both iOS and OS X into a single file and we'd probably need to take that into account for merging it back in csound.

**Comment by kunstmusik** [comment]
Okay, maybe the best thing for the moment is to focus on the dynamic
libraries for this issue, and worry about CsoundObj in another issue or
PR.  I'll respond with more information once I have time to look at
everything that's been done.

**Comment by vlazzarini** [comment]
We have an xcframework now and that seems to be sufficient for iOS development. If any further need arises, a new ticket can be opened.
